group/assignment5/sridha2412.py

```diff
@@ -0,0 +1,71 @@
+from typing import List, Tuple
+
+def common_prefix(word1: str, word2: str) -> int:
+    """ A method to find the common prefix in two words and return the end position
+    Args:
+        word1: first word
+        word2: second word
+    Returns:
+        int: position end of common prefix
+    """
+
+    for i in range(0, min(len(word1), len(word2))):
+        if word1[i] != word2[i]:
+            break
+    return i
+
+
+def create_DAG(words: List[str]) -> dict:
+    """ A method to create a direct acyclic graph from a list of words
+    Args:
+        words: list of words e.g ['ART', 'RAT', 'CAT', 'CAR']
+    Returns:
+        dict: a DAG e.g. {'A': [R], 'R': ['C'], 'C', 'T': ['R']}
+    """
+
+    alphabets = set()
+    for word in words:
+        alphabets = alphabets.union(word)
+
+    graph = {alpha: {"set": set(), "visited": False} for alpha in alphabets}
```

Comment:
Is whether a node is visited really a part of the graph structure?

Reply:
Oh, yeah, I suppose it could be.

Your solution might not need multiple dfs-like function that need a visited counter even with the optionals, so I suppose it doesn't matter either way

Reply:
I do want to complain about the set named "set", though:)

edges, adjacent, whatever, there probably is some more fitting name